dartagnan/src/main/java/com/dat3m/dartagnan/encoding/ExpressionEncoder.java

```diff
@@ -281,6 +281,29 @@ public Formula visitIntUnaryExpression(IntUnaryExpr iUn) {
                 String.format("Encoding of (%s) %s %s not supported.", iUn.getType(), iUn.getKind(), inner));
     }
 
+    @Override
+    public Formula visitIntConcat(IntConcat expr) {
+        final BitvectorFormulaManager bvmgr = bitvectorFormulaManager();
+        final List<Expression> operands = expr.getOperands();
+        BitvectorFormula enc = (BitvectorFormula) operands.get(0).accept(this);
```

Comment:
Neither the constructor nor the `ExpressionFactory` prohibits to pass the empty list as operands, thus the `get` can fail.

Reply:
The constructor prohibits this implicitly right now: an empty input would result in a `bv0` type which is not legal.
Adding an explicit check might be nicer though.